tests/test_frame.py

```diff
@@ -1251,3 +1253,26 @@ def test_boolean_loc() -> None:
     df = pd.DataFrame([[0, 1], [1, 0]], columns=[True, False], index=[True, False])
     check(assert_type(df.loc[True], pd.Series), pd.Series)
     check(assert_type(df.loc[:, False], pd.Series), pd.Series)
+
+
+def test_groupby_result() -> None:
+    # GH 142
+    df = pd.DataFrame({"a": [0, 1, 2], "b": [4, 5, 6], "c": [7, 8, 9]})
+    index, value = next(df.groupby(["a", "b"]).__iter__())
```

Comment:
One small change: a two-element list can also be unpacked into two variables.

I tested the following small change on your branch:

```diff
     Iterable,
+    Iterator,
     List,

...
-    index, value = next(df.groupby(["a", "b"]).__iter__())
+    iterator = df.groupby(["a", "b"]).__iter__()
+    assert_type(iterator, Iterator[Tuple[Tuple, pd.DataFrame]])
+    index, value = next(iterator)
```

Reply:
> One small change: a two-element list can also be unpacked into two variables.
> 
> I tested the following small change on your branch:
> 
> ```diff
>      Iterable,
> +    Iterator,
>      List,
> 
> ...
> -    index, value = next(df.groupby(["a", "b"]).__iter__())
> +    iterator = df.groupby(["a", "b"]).__iter__()
> +    assert_type(iterator, Iterator[Tuple[Tuple, pd.DataFrame]])
> +    index, value = next(iterator)
> ```

OK, that is in the next commit.